### DISCUSSION TITLE
Project8 LIOJ 1049 陣列最短距離的優化解 (python)

### DISCUSSION THREAD

**Discussion by nasutest** [Lidemy/forum#213] [其他問題]
老師你好,

[Project8 LIOJ 1049 陣列最短距離](https://oj.lidemy.com/problem/1049) 
自己怎麼想都無法解出來 
想請教怎麼實現比大小比到後面的數字更大後就停第二個迴圈到上一層

把實際input那段直接改成兩個list做測試
故意把最短距離的第二個數字在第二個list中改到最後一個
試過用abs(nx[i] - ny[j+1])的方式但是這樣比不到第二個數字的最後一個

請老師協助 感恩

```
nx = [1, 10, 13, 15, 18]
ny = [3, 6, 7, 11, 13]

mini_dis = abs(nx[0] - ny[0])
for i in range(len(nx)):
    for j in range(len(ny)):
        com_dis = abs(nx[i] - ny[j])
        if com_dis < mini_dis:
            mini_dis = com_dis
print(mini_dis)
```

**Comment by cwc329** [comment]
想先問一下為何要後面數字更大就停止？
如果是這樣
```
1 2
3
1 4
```
依照你的寫法答案會是 2，
這樣好像就不對了。

**Reply by nasutest** [reply]
因為影片https://lidemy.com/courses/793973/lectures/14680781 
3:50中有說到
比到後面的數字差更大的話就不用再比下去

我的解沒有包含這個

**Reply by cwc329** [reply]
是數字差還是數字？
因為你提問說「數字」我才問這個問題的

**Reply by nasutest** [reply]
我的理解是比較後的「數字差」

**Reply by cwc329** [reply]
如果單純要跳出迴圈
Google: <language> 跳出迴圈
應該就可以了
